lib/IRGen/IRGenFunction.cpp

```diff
@@ -431,3 +431,28 @@ Address IRGenFunction::emitAddressAtOffset(llvm::Value *base, Offset offset,
   auto slotPtr = emitByteOffsetGEP(base, offsetValue, objectTy);
   return Address(slotPtr, objectAlignment);
 }
+
+void IRGenFunction::emitTrap(bool EmitUnreachable) {
+  if (IGM.IRGen.Opts.shouldOptimize()) {
+    // Emit unique side-effecting inline asm calls in order to eliminate
```

Comment:
Shouldn't this be the Doxygen comment for the function?

Reply:
IMO this describes the implementation details at too low a level. The doxygen comment I added in the header file describes the high-level effect more succinctly.